python/lsst/pipe/analysis/visitAnalysis.py

```diff
@@ -550,19 +550,15 @@ def readCatalogs(self, dataRefList, dataset, repoInfo, aliasDictList=None, fakeC
 
             # Compute Focal Plane coordinates for each source if not already there
             if self.config.hasFakes:
-                # Get just a single pixel of the exposure
-                # Exposure is needed for ExposureInfo including bounding
-                # polygon
-                pix = lsstgeom.Box2I(lsstGeom.Point2I(0, 0),
-                                     lsstGeom.Point2I(0, 0))
-                exp = butler.get("calexp_sub", dataId, bbox=pix)
-                wcs = exp.getWcs()
-
-                # Hacky but fast way of getting just the mask
+                # getUri is less safe but enables us to use an efficient
+                # ExposureFitsReader
+                fname = repoInfo.butler.getUri("calexp", dataId)
```

Comment:
Why getUri rather than `_filename`?

Reply:
The former searches parents; the latter doesn't.